### PR TITLE
Fix zoom and tilt for oblique targets

### PR DESCRIPTION
**Pull request by raulvera59**
Also refactor PowerPort a bit.

Draft until tested.

### CODE REVIEW

**Comment by raulvera59** [comment]
Fully tested and debugged so no longer draft.

File: power_port_vision.py [hunk 1]
```diff
@@ -178,41 +176,64 @@ def set_camera_zoom_and_tilt(self):
...
+            if (
+                self.previous_power_port.get_width()
+                > self.previous_power_port.get_height()
+            ):
+                min_from_centre = abs(
+                    self.previous_power_port.get_leftmost_x() - FRAME_WIDTH / 2
+                )
+                max_from_centre = abs(
+                    self.previous_power_port.get_rightmost_x() - FRAME_WIDTH / 2
+                )
```
Comment (Injector-Spenral) [inline power_port_vision.py:205]: Should these be called left and right as opposed to min and max? Makes it seem like one distance is larger.

File: power_port_vision.py [hunk 2]
```diff
@@ -178,41 +176,64 @@ def set_camera_zoom_and_tilt(self):
...
+            # from the minimum or maximum x from the centre of the image, and
+            # the length we want to scale to is the appropriate maximum, clipped
             # to the maximum and a minimum of 1.
-            min_from_centre = self.previous_target_min_x - FRAME_WIDTH / 2
-            max_from_centre = self.previous_target_max_x - FRAME_WIDTH / 2
-            excursion = max(abs(min_from_centre), abs(max_from_centre))
-            new_zoom = round(self.zoom_factor * self.MAX_EXCURSION / excursion, 2)
+            if (
+                self.previous_power_port.get_width()
+                > self.previous_power_port.get_height()
```
Comment (Injector-Spenral) [inline power_port_vision.py:198]: I'm not sure about this, but should it check if the ratio is above/below 4:3?

**Review by Injector-Spenral — APPROVED** [review]
A couple of small questions, otherwise LGTM

File: power_port_vision.py (hunk 2 above)
Comment (raulvera59) [reply power_port_vision.py:198]: Hmmm. Very good observation. I'm not sure what the exact answer is, but this at least prevents the problems we were having with obliques. It's still an improvement, so I'm going to merge it, but I will add a comment with a TODO to come up with a better test.

**Pull request merged**